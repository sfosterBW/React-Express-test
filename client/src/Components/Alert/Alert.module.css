.alert {
  align-items: center;
  background: #ff0100a6;
  border-radius: 0.5vmin;
  display: flex;
  margin: 2vmin;
  padding: 1vmin;
}

.alert button {
  height: 4vmin;
  width: 4vmin;
  margin-left: 2vmin;
}

.alert button:hover {
  cursor: pointer;
}

.alert strong {
  background: white;
  border: 0.5vmin solid black;
  margin-left: 2vmin;
  margin-right: 2vmin;
  padding: 1vmin;
}
