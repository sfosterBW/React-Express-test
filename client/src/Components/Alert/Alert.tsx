import React, { FC } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { RootState } from '../../Reducers/store'
import { toggleAlert } from '../../Reducers/alertReducer'
import styles from './Alert.module.css'

const Alert: FC<{message: string}> = ({ message }) => {
  const selectToggle = (state: RootState) => state.alert.toggle
  const toggle = useSelector(selectToggle)
  const dispatch = useDispatch()

  if(!toggle) {
    return null
  }

  return (
    <div className={styles.wrapper} data-testid="alert-wrapper">
      <section className={styles.alert}>
        <strong>!!!</strong>
        <p className={styles.message}>
          {message}
        </p>
        <button
          className={styles.button}
          data-testid="close-button"
          onClick={() => {dispatch(toggleAlert(message, !toggle))}}
        >
          &times;
        </button>
      </section>
    </div>
  )
}

export default Alert
