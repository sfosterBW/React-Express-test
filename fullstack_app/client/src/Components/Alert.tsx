import React, { FC } from 'react'

interface Props {
  onClose: () => void
  title: string
}

const Alert: FC<Props> = ({ onClose, title }) => {
  return (
    <div>
      <p>{title}</p>
      <button onClick={onClose}>&times;</button>
    </div>
  )
}

export default Alert
