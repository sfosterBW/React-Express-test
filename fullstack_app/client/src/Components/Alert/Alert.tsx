import React, { FC } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { RootState } from '../../utils/store'
import { toggleAlert } from '../../utils/actions'
import styles from './Alert.module.css'

const Alert: FC = () => {
  const selectToggle = (state: RootState) => state.alert.toggle
  const toggle = useSelector(selectToggle)
  const dispatch = useDispatch()

  if(!toggle) {
    return null
  }

  return (
    <div className={styles.wrapper}>
      <section className={styles.alert}>
        <p className={styles.message}>{String(toggle)}</p>
        <button
          className={styles.button}
          onClick={() => {dispatch(toggleAlert(!toggle))}}
        >
          &times;
        </button>
      </section>
    </div>
  )
}

export default Alert
