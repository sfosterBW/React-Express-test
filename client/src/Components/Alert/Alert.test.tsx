import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import Alert from './Alert'

const value = true
const mockDispatch = jest.fn().mockReturnValue(!value)
jest.mock('react-redux', () => ({
  useSelector: jest.fn().mockReturnValue(true),
  useDispatch: () => mockDispatch
}))

describe('the Alert component', () => {
  const message = "test"
  const component = <Alert message={message} />

  it('renders on mount functions as expected', () => {
    const { getByTestId } = render(component)
    fireEvent.click(getByTestId('close-button'))
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })

  it('renders the same as last time with props', () => {
    const { container } = render(component)
    expect(container).toMatchSnapshot()
  })
})
