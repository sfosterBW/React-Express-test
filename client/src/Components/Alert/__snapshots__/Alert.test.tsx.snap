// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Alert component renders the same as last time with props 1`] = `
<div>
  <div
    class="wrapper"
    data-testid="alert-wrapper"
  >
    <section
      class="alert"
    >
      <strong>
        !!!
      </strong>
      <p
        class="message"
      >
        test
      </p>
      <button
        class="button"
        data-testid="close-button"
      >
        Ã—
      </button>
    </section>
  </div>
</div>
`;
